File: 07.function/1065_arithmetic_sequence.py
#!/usr/bin/env python

n = int(input())
cnt = 0

for i in range(1,n+1):
    if i < 100:
        cnt += 1
    else:
        nums = [int(c) for c in str(i)]
        d = nums[0] - nums[1]
        check = True
        for k in range(1,len(nums)-1):
            if d != nums[k] - nums[k+1]:
                check = False
                break

        if check:
            cnt += 1

print(cnt)

File: 07.function/1065_arithmetic_sequence.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    int n, d;
    int cnt=0;
    bool check;
    string s;
    scanf("%d", &n);

    for (int a=1; a<=n; ++a) {
        if (a < 100)
            cnt++;
        else {
            s = to_string(a);
            d = int(s[0]) - int(s[1]);
            check = true;
            for (int i=1; i<s.size()-1; ++i) {
                if (d != int(s[i]) - int(s[i+1])) {
                    check = false;
                    break;
                }
            }
            if (check) cnt++;
        }
    }
    cout << cnt << endl;

    return 0;
}
